### ISSUE TITLE
Problem with launching a cc2.8xlarge instance

### ISSUE THREAD

**Issue by izharw** [jtriley/StarCluster#274]
Hello,

I'm trying to start a single-node cluster using a cc2.8xlarge instance. I can see on the EC2 web console that the instance is getting initialized but terminates after a short period (before being fully initialized). If I try to start an instance with the starcluster-HVM image through the EC2 web page everything works well. Hence, I suspect it has something to do with the starcluster initialization sequence. Oddly, when I'm using another HVM-based instance such as m3.xlarge everything seems to be working fine.

My configuration is the following:
[cluster test]
KEYNAME = chematria_oregon
CLUSTER_SIZE = 1
CLUSTER_USER = ec2user
CLUSTER_SHELL = bash
NODE_IMAGE_ID = ami-486afe78
NODE_INSTANCE_TYPE = cc2.8xlarge
PERMISSIONS = ssh

Below is the output of the  'starcluster start -s 1 -c test test' command:

> > > Starting cluster...
> > > Launching a 1-node cluster...
> > > 2013-07-18 16:11:12,698 cluster.py:1016 - DEBUG - Launching master (ami: ami-486afe78, type: cc2.8xlarge)
> > > Creating security group @sc-test...
> > > Creating placement group @sc-test...
> > > 2013-07-18 16:11:17,141 cluster.py:783 - DEBUG - Userdata size in KB: 0.46
> > > Reservation:r-ce78f7f9
> > > Waiting for cluster to come up... (updating every 30s)
> > > 2013-07-18 16:11:49,282 cluster.py:693 - DEBUG - existing nodes: {}
> > > 2013-07-18 16:11:49,283 cluster.py:701 - DEBUG - adding node i-d639b6e1 to self._nodes list
> > > 2013-07-18 16:11:50,109 cluster.py:709 - DEBUG - returning self._nodes = [<Node: master (i-d639b6e1)>]
> > > Waiting for all nodes to be in a 'running' state...
> > > 2013-07-18 16:11:50,320 cluster.py:693 - DEBUG - existing nodes: {u'i-d639b6e1': <Node: master (i-d639b6e1)>}
> > > 2013-07-18 16:11:50,320 cluster.py:696 - DEBUG - updating existing node i-d639b6e1 in self._nodes
> > > 2013-07-18 16:11:50,321 cluster.py:709 - DEBUG - returning self._nodes = [<Node: master (i-d639b6e1)>]
> > > 2013-07-18 16:12:20,684 cluster.py:693 - DEBUG - existing nodes: {}    |   0% 
> > > 2013-07-18 16:12:20,685 cluster.py:709 - DEBUG - returning self._nodes = []
> > > 1/1 |||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||| 100% 
> > > !!! ERROR - No active cluster nodes found!

I wonder if anybody noticed a similar problem in launching cc2.8xlarge instances?

Thanks,
Izhar

**Comment by izharw** [comment]
I forgot to mention that I'm using the latest startcluster version from Git.

Izhar

**Comment by FinchPowers** [comment]
Have a look at that pull request
https://github.com/jtriley/StarCluster/pull/163

**Comment by jtriley** [comment]
@izharw Thanks for reporting. I'm able to reproduce this and am debugging the issue now.

@FinchPowers This isn't related to #163 - this has to do with the patch to add all ephemeral drives to the block device mapping.

**Comment by jtriley** [comment]
This appears to only be a problem in us-west-2. If I launch a cc2.8xlarge in us-east-1 using the 12.04 HVM AMI it works fine. This should be fun...

**Comment by jtriley** [comment]
I also don't have this problem in eu-west-1 region...

**Comment by jtriley** [comment]
Looking closer at this it seems the 'real' error from this is:

```
Server.InsufficientInstanceCapacity: Insufficient capacity to satisfy instance request
```

@izharw Can you please double-check that this is the case for you as well? You can double check using the AWS EC2 console and look at the details for any of your terminated instances.

I'm beginning to think this is a transient issue with EC2 in the us-west-2 region given that the error states an insufficient capacity issue and also given the fact that removing everything related to block device maps and userdata scripts when requesting the instances still produces this issue for me. Also, using cc2.8xlarge with 12.04 HVM AMIs in the other cluster-compute-supported regions (currently us-east-1 and eu-west-1) works fine _with_ all of the latest block device mapping and userdata stuff.

With that said I have a couple more things to experiment with before I close this issue for good.

**Comment by jtriley** [comment]
OK, I think I just figured it out. This is _not_ an issue with block device mappings or userdata scripts. This has to do with the fact that StarCluster automatically creates a placement group and launches the instances in that placement group provided the instance type supports placement groups. 

So in this case there _are_ cc2.8xlarge instances available, however, there are not enough resources available to _also_ launch them into a placement group at this time in us-west-2. This is a transient issue that will certainly go away with time, however, it's probably best to make placement groups an optional thing that must be explicitly chosen by users rather than automatically created and used depending on the instance type...

**Comment by izharw** [comment]
Yes. I'm getting the "Server. InsufficientInstanceCapacity: Insufficient
capacity to satisfy instance request" error message too.

Thanks,
Izhar

**Comment by FinchPowers** [comment]
See https://github.com/jtriley/StarCluster/pull/276

**Comment by izharw** [comment]
Thanks! I migrated my data meanwhile to the us-east-1 region (everything
worked well over there. A bit more expensive though...). I'll try using the
patch next time I'm launching a large batch of jobs.
